source/core/slang-string.h

```diff
@@ -315,6 +315,15 @@ namespace Slang
 
             return cloneWithCapacity(newCapacity);
         }
+
+            /// Overload delete to silence ASAN new-delete-type-mismatch errors.
+            /// These occur because the allocation size of StringRepresentation
+            /// does not match deallocation size (due variable sized string payload).
+        void operator delete(void* p)
```

Comment:
We could wrap this in a `#ifdef __SANITIZE_ADDRESS__` to only have it when ASAN is enabled.